### ISSUE TITLE
linter/prefer-for-of: False positive

### ISSUE THREAD

**Issue by DreierF** [oxc-project/oxc#13200]
The following code produces a false positive that is not produced by typescript-eslint
```
type Series = { data: Array<{ x: number; y: number }> };

function a(series: Series, substract: Series[]): void {
	for (let x = 0; x < series.data.length; x++) { // Produces prefer-for-of FP
		let newValue = series.data[x]!.y;
		for (const otherSeries of substract) {
			newValue -= otherSeries.data[x]!.y; // Notice that x is used here to reference another array, so for-of won't work
		}
		series.data[x]!.y = newValue;
	}
}
```

Produces:
```
  × typescript-eslint(prefer-for-of): Expected a `for-of` loop instead of a `for` loop with this simple iteration.
     ╭─[<file>]
 175 │     // Produces prefer-for-of
 176 │     for (let x = 0; x < series.data.length; x++) {
     ·          ──────────────────────────────────────
 177 │         let newValue = series.data[x]!.y;
     ╰────
  help: Consider using a for-of loop for this simple iteration.
```
